sql/snapshot.sql

```diff
@@ -0,0 +1,546 @@
+	created_at timestamp DEFAULT now() NOT NULL,
+	updated_at timestamp DEFAULT now() NOT NULL,
+	CONSTRAINT metagraph_snapshot_pk PRIMARY KEY (metagraph_id, hash),
+	CONSTRAINT metagraph_snapshot_unique UNIQUE (metagraph_id, ordinal),
+	CONSTRAINT metagraph_snapshots_global_snapshots_fk FOREIGN KEY (global_snapshot_hash) REFERENCES global_snapshots(hash)
+);
+
+-- Table Triggers
+
+create trigger set_updated_at_metagraph_snapshot before
+update
+    on
+    public.metagraph_snapshots for each row execute function update_updated_at_column();
+
+
+-- public.metagraph_spend_transactions definition
+
+-- Drop table
+
+-- DROP TABLE metagraph_spend_transactions;
+
+CREATE TABLE metagraph_spend_transactions (
+	metagraph_id varchar NOT NULL,
+	allow_spend varchar NULL,
+	CONSTRAINT destination_addr_not_null CHECK ((destination_addr IS NOT NULL)),
+	CONSTRAINT metagraph_spend_transactions_pk PRIMARY KEY (hash),
+	CONSTRAINT original_not_null CHECK ((original IS NOT NULL)),
+	CONSTRAINT metagraph_spend_transactions_metagraph_allow_spends_fk FOREIGN KEY (allow_spend) REFERENCES metagraph_allow_spends(hash)
+)
+INHERITS (public.abstract_transactions);
+
+-- Table Triggers
+
+create trigger set_updated_at_metagraph_spend_transactions before
+update
+    on
+    public.metagraph_spend_transactions for each row execute function update_updated_at_column();
+
+
+-- public.metagraph_token_locks definition
+
+-- Drop table
+
+-- DROP TABLE metagraph_token_locks;
+
+CREATE TABLE metagraph_token_locks (
+	ordinal int8 NOT NULL,
+	metagraph_id varchar NOT NULL,
+	unlock_epoch int8 NOT NULL,
+	CONSTRAINT metagraph_token_locks_pk PRIMARY KEY (hash),
+	CONSTRAINT metagraph_token_locks_unique UNIQUE (hash, ordinal),
+	CONSTRAINT source_addr_not_null CHECK ((source_addr IS NOT NULL))
+)
+INHERITS (public.abstract_transactions);
+
+-- Table Triggers
+
+create trigger set_updated_at_metagraph_token_locks before
+update
+    on
+    public.metagraph_token_locks for each row execute function update_updated_at_column();
+
+
+-- public.metagraph_token_unlocks definition
+
+-- Drop table
+
+-- DROP TABLE metagraph_token_unlocks;
+
+CREATE TABLE metagraph_token_unlocks (
+	lock_reference_ordinal int8 NOT NULL,
+	lock_reference_hash varchar NOT NULL,
+	amount int8 NOT NULL,
+	metagraph_id varchar NOT NULL,
+	address varchar NOT NULL,
+	created_at timestamp DEFAULT now() NOT NULL,
+	updated_at timestamp DEFAULT now() NOT NULL,
+	CONSTRAINT metagraph_token_unlocks_pk PRIMARY KEY (lock_reference_ordinal, lock_reference_hash),
+	CONSTRAINT metagraph_token_unlocks_token_locks_fk FOREIGN KEY (lock_reference_hash,lock_reference_ordinal) REFERENCES metagraph_token_locks(hash,ordinal)
+);
+
+-- Table Triggers
+
+create trigger set_updated_at_metagraph_token_unlocks before
+update
+    on
+    public.metagraph_token_unlocks for each row execute function update_updated_at_column();
+
+
+-- public.proofs definition
+
+-- Drop table
+
+-- DROP TABLE proofs;
+
+CREATE TABLE proofs (
```

Comment:
How about `snapshot_proofs` since this is specifically signatures for snapshots, not some other resource.